networkit/graph.pyx

```diff
@@ -612,6 +612,37 @@ cdef class Graph:
 			yield dereference(it)
 			preincrement(it)
 
+	def iterNeighborsWeights(self, u):
+		"""
+		Iterates over a range of the neighbors of a node including the edge weights.
+		The iterator is not safe to use with unweighted graphs. To avoid unsafe behavior
+		a runtime error will be thrown.
+
+		Parameters:
+		-----------
+		u : Node
+		"""
+		if not self._this.isWeighted():
+			raise RuntimeError("iterNeighborsWeights: Use this iterator only on weighted graphs.")
+		
+		it = self._this.weightNeighborRange(u).begin()
+		while it != self._this.weightNeighborRange(u).end():
+			yield dereference(it)
+			preincrement(it)		
+
+	def iterInNeighborsWeights(self, u):
+		"""
+		Iterates over a range of the in-neighbors of a node including the edge weights.
+
+		Parameters:
+		-----------
+		u : Node
+		"""
```

Comment:
Here, the same check should be added - or is this handled in C++?

Reply:
Yes, this also needs it.